go/sentry/sentry.go

```diff
@@ -16,10 +18,15 @@ import (
 var _ api.Backend = (*backend)(nil)
 
 type backend struct {
+	sync.RWMutex
+
 	logger *logging.Logger
 
 	consensus consensus.Backend
 	identity  *identity.Identity
+
+	upstreamTLSCertificate     *tls.Certificate
```

Comment:
I don't think you actually need to separate between the two here as you add them to a pool later. You could just simplify into `upstreamTLSCertificates [][]byte`.